src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs

```diff
@@ -3339,8 +3315,16 @@ private IEnumerable<SourceRepository> GetEffectiveSources(IEnumerable<SourceRepo
         {
             // Always have to add the packages folder as the primary repository so that
             // dependency info for an installed package that is unlisted from the server is still available :(
-            var effectiveSources = new List<SourceRepository>(primarySources);
-            effectiveSources.Add(PackagesFolderSourceRepository);
+            var count = (primarySources?.Count() ?? 0) +
```

Comment:
this new code is confusing... any particular reason?

Reply:
initializing a list with a fixed size can be efficient. You can read about it [here](https://msdn.microsoft.com/en-us/library/dw8e0z9z.aspx#Anchor_2).

Reply:
same as i mentioned above. Do you really think it ever increase the default size of this list?